3p/messaging.js

```diff
@@ -79,7 +84,9 @@ function startListening(win) {
     // Parse JSON only once per message.
     const data = /** @type {!Object} */ (
         JSON.parse(event.data.substr(4)));
-    if (data.sentinel != win.context.amp3pSentinel) {
+    if (sentinelNameChange && data.sentinel != win.context.sentinel) {
+      return;
+    } else if (data.sentinel != win.context.amp3pSentinel) {
```

Comment:
This needs to check `!sentinelNameChange`. As is, this will always fail when using the sentinel-name-change experiment is enabled.